HoloStorageAccessor/internal/openapi/utils.go

```diff
@@ -0,0 +1,154 @@
+package openapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type HologramQueryDetails struct {
+	IDs          []string
+	Mode         string
+	CreationMode string
+}
+
+type HologramPostInput struct {
+	Hologram Hologram
+	Author   Author
+	Patient  Patient
+}
+
+func ParseHologramUploadPostInput(formData url.Values) (HologramPostInput, error) {
+	var authorData Author
+	var patientData Patient
+	var hologramData Hologram
+
+	for key, _ := range formData {
+		switch key {
+		case "patient":
+			err := json.Unmarshal([]byte(formData.Get(key)), &patientData)
+			if err != nil {
+				return HologramPostInput{}, errors.New("Unable to prase patient data")
+			}
+			if patientData.Pid == "" {
+				return HologramPostInput{}, errors.New("Patient ID is required")
+			}
+			hologramData.Pid = patientData.Pid
+		case "author":
+			err := json.Unmarshal([]byte(formData.Get(key)), &authorData)
+			if err != nil {
+				return HologramPostInput{}, errors.New("Unable to prase author data")
+			}
+			if authorData.Aid == "" {
+				return HologramPostInput{}, errors.New("Author ID is required")
+			}
+			hologramData.Aid = authorData.Aid
+		case "creationDate":
+			creationDate, err := time.Parse(time.RFC3339, formData.Get(key))
+			if err != nil {
+				return HologramPostInput{}, errors.New(key + " is not in RFC3339 standards")
```

Comment:
Probably `formData.Get(key)` will be the more helpful output (same for the similar lines below)

Reply:
Changed both instances to return both the key and value in the error message